### PR TITLE
Update navstar to fixed ipvs

### PR DESCRIPTION
**Pull request by aeyakovenko**
## High Level Description

Fixes a bug in minuteman routes that prevents VIPs from working in bridge mode.

## Related Issues

  - [DCOS-13228](https://mesosphere.atlassian.net/browse/DCOS-13228) TRACKING: dcosjira-13228, VIPs are not working in bridge mode

## Checklist for all PR's

  - [ ] Included a test which will fail if code is reverted but test is not. If there is no test please explain here:
Added networking tests to cover every permunation of VIP
  - [ ] Read the [DC/OS contributing guidelines](https://github.com/dcos/dcos/blob/master/contributing.md)
  - [ ] Followed relevant code rules [Rules for Packages and Systemd](https://github.com/dcos/dcos/tree/master/docs)

## Checklist for component/package updates:

If you are changing components or packages in DC/OS (e.g. you are bumping the sha or ref of anything underneath `packages`), then in addition to the above please also include:

  - [ ] Change log from the last version integrated (this should be a link to commits for easy verification and review):
https://github.com/dcos/minuteman/compare/452b57b07d08def9a4586b6be046943785440529...fe299aa5c9e7956407643f2284c91fd2c52ac955
https://github.com/dcos/navstar/compare/4ef753e175a93adc597827aa28a836e43b8ccf3d...439b52afd04cd0614d52b9c80c18835fcf916cf2
  - [ ] Test Results: [link to CI job test results for component]
https://circleci.com/gh/dcos/minuteman/424?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link
  - [ ] Code Coverage (if available): [link to code coverage report]
https://codecov.io/gh/dcos/minuteman/commit/47b47a162da51285724c7a80b29dc5d8cc07db5f
___
**PLEASE FILL IN THE TEMPLATE ABOVE** / **DO NOT REMOVE ANY SECTIONS ABOVE THIS LINE**

## Instructions and Review Process

**What is the review process and when will my changes land?**

All PRs require 2 approvals using github's [pull request reviews](https://help.github.com/articles/about-pull-request-reviews/).

Reviewers should be:
* Developers who understand the code being modified.
* Developers responsible for code that interacts with or depends on the code being modified.

It is best to proactively ask for 2 reviews by @mentioning the candidate reviewers in the PR comments area. The responsibility is on the developer submitting the PR to follow-up with reviewers and make sure a PR is reviewed in a timely manner. Once a PR has **2 ship-it's**, **no red reviews**, and **all tests are green** it will be included in the [next train](https://github.com/dcos/dcos/blob/master/contributing.md).

### CODE REVIEW

**Comment by asridharan** [comment]
@aeyakovenko looks like code quality issues. Might want to run flake on that file in a container before submitting.

**Comment by aeyakovenko** [comment]
@asridharan can you add @GoelDeepak and @mellenburg as reviewers?

**Comment by asridharan** [comment]
@aeyakovenko once this ready for review just remove the [WIP] tag and ping anyone you want to review this.

File: packages/dcos-integration-test/extra/test_networking.py [hunk 1]
```diff
@@ -27,9 +30,120 @@ def _ensure_routable():
...
+        returned_uuid = ensure_routable(cmd, sp)()
+        logging.info('returned_uuid is: {}'.format(returned_uuid))
+    assert(returned_uuid is not None)
+    assert(returned_uuid['test_uuid'] == app_uuid)
+    logging.info('VIP_TEST PASSED {}'.format(teststr))
+    return app_uuid
+
+
+@pytest.mark.skipif(not lb_enabled(), reason='Load Balancer disabled')
+def test_vip(dcos_api_session):
```
Comment (orsenthil) [inline packages/dcos-integration-test/extra/test_networking.py]: This should see some overhaul.

1) Keep `addrs` and one of `vipnets` and `proxynets` (call it something common).
2) Have only two external for-loops to track. Kill the outer for-loop to iterate over samehosts  [True, False]. See if this can folded inside the code.
3) call the variable names failed, passed, skipped something with a more meaningful identifer (easy).
4) See if using of `num` can be avoided. (Encode the ports within).

5) Ideally, if this is constructing the configurations of the ip-addresses, just construct them early and separately as a structure by itself into a list and then iterate over it to add to the threads. ?

File: packages/dcos-integration-test/extra/test_networking.py [hunk 2]
```diff
@@ -16,7 +19,7 @@ def lb_enabled():
     return config['enable_lb'] == 'true'
 
 
-def ensure_routable(cmd, service_points, timeout=120):
+def ensure_routable(cmd, service_points, timeout=180):
```
Comment (mellenburg) [inline packages/dcos-integration-test/extra/test_networking.py]: This function should be fixed up. It makes no sense to return a callable function in this usage context. Rather, get ride of _ensure_routable entirely and just put this decorator at the top level. Thus, you just need to call ensure_routable(cmd, service_points) to evaluate this and nothing will be lost

File: packages/dcos-integration-test/extra/test_networking.py [hunk 3]
```diff
@@ -27,9 +30,123 @@ def _ensure_routable():
...
+                                                                           proxynet)
+    logging.info('VIP_TEST START {}'.format(teststr))
+    agents = list(dcos_api_session.slaves)
+    # make sure we can reproduce
+    random.seed(vip)
+    random.shuffle(agents)
+    host1 = agents[0]
+    host2 = agents[0]
+    if not samehost:
+        host2 = agents[1]
```
Comment (mellenburg) [inline packages/dcos-integration-test/extra/test_networking.py]: This test will likely fail for any cluster that has more than 2 private agents. Either this needs to be written in a way that takes the service points from the app deployment to figure out where to ping or this whole test needs to be skipped when we have more than 2 private agents. I would vote for the former. Any reason that's not possible?

File: packages/dcos-integration-test/extra/test_networking.py [hunk 4]
```diff
@@ -27,9 +30,123 @@ def _ensure_routable():
...
+    return app_uuid
+
+
+@pytest.mark.skipif(not lb_enabled(), reason='Load Balancer disabled')
+def test_vip(dcos_api_session):
+    """Test every permutation of VIP
+    """
+    num = 0
+    addrs = [['1.1.1.{}:{}', '1.1.1.{}:{}'],
+             ['/namedvip{}:{}', 'namedvip{}.marathon.l4lb.thisdcos.directory:{}']]
```
Comment (mellenburg) [inline packages/dcos-integration-test/extra/test_networking.py]: @aeyakovenko I noticed you changed this from parametrize. That was the preferred way of configuring something like this. Did changing away from that fix that app cleanup issue? Or is app cleanup still a problem?

File: packages/dcos-integration-test/extra/test_networking.py (hunk 3 above)
Comment (aeyakovenko) [inline packages/dcos-integration-test/extra/test_networking.py]: @mellenburg can you explain why it will fail?  it should run with more than two private agents, and distribute across all the private agents randomly.

File: packages/dcos-integration-test/extra/test_networking.py (hunk 4 above)
Comment (aeyakovenko) [inline packages/dcos-integration-test/extra/test_networking.py]: you mean the parameter fixtures?  There is no way to run all those tests in parallel.  If run sequentially they take 40 minutes :).

Comment (mellenburg) [inline packages/dcos-integration-test/extra/test_networking.py]: Ah, yes. Of course.

The idea of using a fixture to start a thread pool like you describe won't really work within the pytest framework because what you would end up having (if I understand this correctly) would be:
```
@pytest.fixture
def my_thread_control():
    # generate all your threads and start them
    yield
    # join your threads and check them
```
and all of your asserting (i.e. the test cases) will be done after joining threads. Anything you define in a test case won't be able to get back into a fixture.

However, I think as we get more and more tests, this issue of parallelization will begin concern more people. I am thinking of building a plugin where you could mark tests like so:
```
@parallel(scope='networking_test')
def test_my_network(some_setup_fixture):
    pass

@parallel(scope='networking_test')
def test_my_network_2(some_setup_fixture):
    pass
```
this would require changing up the runtest protocol, but pytest is really open about this, so I dont think it will be that difficult.

I think for now we should probably just stick to making the huge lot of threads in a single test as you do now, so to keep this focused on the test instead of pytest madness

File: packages/dcos-integration-test/extra/test_networking.py (hunk 3 above)
Comment (mellenburg) [inline packages/dcos-integration-test/extra/test_networking.py]: Oops, brainfart. No problem here. My bad

File: packages/dcos-integration-test/extra/test_networking.py [hunk 5]
```diff
@@ -27,9 +30,123 @@ def _ensure_routable():
...
+    logging.info('host1 is is: {}'.format(host1))
+    logging.info('host2 is is: {}'.format(host2))
+    origin_app, app_uuid = get_test_app_in_docker()
+    origin_app['container']['docker']['portMappings'][0]['labels'] = {'VIP_0': vip}
+    origin_app['container']['docker']['network'] = vipnet
+    if vipnet == 'USER':
+        origin_app['ipAddress'] = {'networkName': 'dcos'}
+    if vipnet == 'HOST':
+        origin_app['cmd'] = '/opt/mesosphere/bin/dcos-shell python '
+        origin_app['cmd'] += '/opt/mesosphere/active/dcos-integration-test/util/python_test_server.py '
```
Comment (mellenburg) [inline packages/dcos-integration-test/extra/test_networking.py]: Can we remove this string-append to a key-value syntax? Please just use:
```
origin_app['cmd'] = 'wow, this command is really really really '  \
    'long. I don't think I will be able to fit it on one line, so I guess ' \
    'its time to use the escape char'
```

File: packages/dcos-integration-test/extra/test_networking.py [hunk 6]
```diff
@@ -27,9 +30,123 @@ def _ensure_routable():
...
+        origin_app['cmd'] += '$PORT0'
+        origin_app['container']['docker']['portMappings'][0]['hostPort'] = 0
+        origin_app['container']['docker']['portMappings'][0]['containerPort'] = 0
+        origin_app['container']['docker']['portMappings'][0]['labels'] = {}
+        origin_app['portDefinitions'] = [{'labels': {'VIP_0': vip}}]
+
+    origin_app['constraints'] = [['hostname', 'CLUSTER', host1]]
+    origin_app['constraints'] = [['hostname', 'CLUSTER', host1]]
```
Comment (mellenburg) [inline packages/dcos-integration-test/extra/test_networking.py]: repeated line; remove

File: packages/dcos-integration-test/extra/test_networking.py [hunk 7]
```diff
@@ -27,9 +30,123 @@ def _ensure_routable():
...
+
+    proxy_app = get_test_app_in_docker()[0]
+    proxy_app['container']['docker']['network'] = proxynet
+    proxy_app['constraints'] = [['hostname', 'CLUSTER', host2]]
+    if proxynet == 'USER':
+        proxy_app['ipAddress'] = {'networkName': 'dcos'}
+    if proxynet == 'HOST':
+        proxy_app['cmd'] = '/opt/mesosphere/bin/dcos-shell python '
+        proxy_app['cmd'] += '/opt/mesosphere/active/dcos-integration-test/util/python_test_server.py '
+        proxy_app['cmd'] += '$PORT0'
```
Comment (mellenburg) [inline packages/dcos-integration-test/extra/test_networking.py]: see above on formatting at once instead of appending

File: packages/dcos-integration-test/extra/test_networking.py [hunk 8]
```diff
@@ -27,9 +30,123 @@ def _ensure_routable():
...
+    returned_uuid = None
+
+    with contextlib.ExitStack() as stack:
+        stack.enter_context(dcos_api_session.marathon.deploy_and_cleanup(origin_app))
+        sp = stack.enter_context(dcos_api_session.marathon.deploy_and_cleanup(proxy_app))
+        cmd = '/opt/mesosphere/bin/curl -s -f -m 5 http://{}/test_uuid'.format(vipaddr)
+        returned_uuid = ensure_routable(cmd, sp)()
+        logging.info('returned_uuid is: {}'.format(returned_uuid))
+    assert(returned_uuid is not None)
+    assert(returned_uuid['test_uuid'] == app_uuid)
```
Comment (mellenburg) [inline packages/dcos-integration-test/extra/test_networking.py]: parentheses are not required for assert and are omitted everywhere else in testing

File: test_util/test_upgrade_vpc.py [hunk 9]
```diff
@@ -211,7 +211,7 @@ def wait_for_dns(hostname):
         # Skip raising an exception on DNS failure so that other tests can run.
         # DNS failure is being tracked at
         # https://mesosphere.atlassian.net/browse/DCOS-13426.
-        #raise Exception(message)  # noqa
+        # raise Exception(message)  # noqa
```
Comment (mellenburg) [inline test_util/test_upgrade_vpc.py]: Can we remove that silly # noqa You did the right syntax fix here, but it wasnt picked up before due to that # noqa

Comment (branden) [inline test_util/test_upgrade_vpc.py]: The missing space here was deliberate -- I wanted this to look like commented-out code, not a block comment. That's what the `# noqa` is there to protect.

Comment (orsenthil) [inline test_util/test_upgrade_vpc.py]: wow. interesting. :)

Comment (mellenburg) [inline test_util/test_upgrade_vpc.py]: Oh, I see now. Yeah, that's fine

Comment (aeyakovenko) [inline test_util/test_upgrade_vpc.py]: code check fails without it.  should i revert?

Comment (mellenburg) [inline test_util/test_upgrade_vpc.py]: Apparently your python linter doesnt respect ```# noqa``` which prevents any linting errors from that line

File: packages/dcos-integration-test/extra/test_networking.py [hunk 10]
```diff
@@ -16,18 +19,131 @@ def lb_enabled():
...
-                    stop_max_delay=timeout * 1000,
-                    retry_on_result=lambda ret: ret is False,
-                    retry_on_exception=lambda x: True)
-    def _ensure_routable():
-        proxy_uri = 'http://{}:{}/run_cmd'.format(service_points[0].host, service_points[0].port)
-        logging.info('Sending {} data: {}'.format(proxy_uri, cmd))
-        r = requests.post(proxy_uri, data=cmd)
-        logging.info('Requests Response: %s', repr(r.json()))
-        assert r.json()['status'] == 0
-    return _ensure_routable
```
Comment (mellenburg) [inline packages/dcos-integration-test/extra/test_networking.py]: Much better. Thanks.

File: packages/dcos-integration-test/extra/test_networking.py [hunk 11]
```diff
@@ -16,18 +19,131 @@ def lb_enabled():
...
+
+    with contextlib.ExitStack() as stack:
+        stack.enter_context(dcos_api_session.marathon.deploy_and_cleanup(origin_app))
+        sp = stack.enter_context(dcos_api_session.marathon.deploy_and_cleanup(proxy_app))
+        cmd = '/opt/mesosphere/bin/curl -s -f -m 5 http://{}/test_uuid'.format(r.vipaddr)
+        returned_uuid = ensure_routable(cmd, sp)
+        logging.info('returned_uuid is: {}'.format(returned_uuid))
+    assert returned_uuid is not None
+    assert returned_uuid['test_uuid'] == app_uuid
+    r.log('PASSED')
```
Comment (mellenburg) [inline packages/dcos-integration-test/extra/test_networking.py]: This log statement is redundant with the pass statement below; please remove

File: test_util/test_upgrade_vpc.py (hunk 9 above)
Comment (aeyakovenko) [inline test_util/test_upgrade_vpc.py]: so revert?

File: packages/dcos-integration-test/extra/test_networking.py (hunk 11 above)
Comment (aeyakovenko) [inline packages/dcos-integration-test/extra/test_networking.py]: it makes watching the build.log live more exciting!

**Comment by asridharan** [comment]
@aeyakovenko any reason why the AWS and Azure tests are failing?

File: packages/dcos-integration-test/extra/test_networking.py [hunk 12]
```diff
@@ -1,37 +1,156 @@
...
-                    retry_on_exception=lambda x: True)
-    def _ensure_routable():
-        proxy_uri = 'http://{}:{}/run_cmd'.format(service_points[0].host, service_points[0].port)
-        logging.info('Sending {} data: {}'.format(proxy_uri, cmd))
-        r = requests.post(proxy_uri, data=cmd)
-        logging.info('Requests Response: %s', repr(r.json()))
-        assert r.json()['status'] == 0
-    return _ensure_routable
+@retrying.retry(wait_fixed=2000,
+                stop_max_delay=TIMEOUT * 1000,
```
Comment (mellenburg) [inline packages/dcos-integration-test/extra/test_networking.py]: Timeout has changed from 2 minutes to 8 minutes and 20 seconds. This seems like a pretty big change. If we need to bump it, maybe 5 minutes (300) would be a more suitable value?

File: packages/dcos-integration-test/extra/test_networking.py [hunk 13]
```diff
@@ -96,21 +188,58 @@ def test_ip_per_container(dcos_api_session):
...
+
+
+@retrying.retry(wait_fixed=2000,
+                stop_max_delay=100 * 2000,
+                retry_on_exception=lambda x: True)
+def geturl(url):
+    rs = requests.get(url)
+    assert rs.status_code == 200
+    r = rs.json()
+    logging.info('geturl {} -> {}'.format(url, r))
```
Comment (mellenburg) [inline packages/dcos-integration-test/extra/test_networking.py]: make this into a debug statement

File: packages/dcos-integration-test/extra/test_networking.py [hunk 14]
```diff
@@ -1,37 +1,156 @@
...
+    agents = list(dcos_api_session.slaves)
+    # make sure we can reproduce
+    random.seed(r.vip)
+    random.shuffle(agents)
+    host1 = agents[0]
+    host2 = agents[0]
+    if not r.samehost:
+        host2 = agents[1]
+    logging.info('host1 is is: {}'.format(host1))
+    logging.info('host2 is is: {}'.format(host2))
```
Comment (mellenburg) [inline packages/dcos-integration-test/extra/test_networking.py]: debug statements please

File: packages/dcos-integration-test/extra/test_networking.py [hunk 15]
```diff
@@ -1,37 +1,156 @@
...
-        logging.info('Requests Response: %s', repr(r.json()))
-        assert r.json()['status'] == 0
-    return _ensure_routable
+@retrying.retry(wait_fixed=2000,
+                stop_max_delay=TIMEOUT * 1000,
+                retry_on_result=lambda ret: ret is False,
+                retry_on_exception=lambda x: True)
+def ensure_routable(cmd, service_points):
+    proxy_uri = 'http://{}:{}/run_cmd'.format(service_points[0].host, service_points[0].port)
+    logging.info('Sending {} data: {}'.format(proxy_uri, cmd))
```
Comment (mellenburg) [inline packages/dcos-integration-test/extra/test_networking.py]: debug statement please

File: packages/dcos-integration-test/extra/test_networking.py [hunk 16]
```diff
@@ -1,37 +1,156 @@
...
+    r = requests.post(proxy_uri, data=cmd)
+    logging.info('Requests Response: %s', repr(r.json()))
```
Comment (mellenburg) [inline packages/dcos-integration-test/extra/test_networking.py]: debug statement please

File: packages/dcos-integration-test/extra/test_networking.py [hunk 17]
```diff
@@ -1,37 +1,156 @@
...
 import requests
-
 import retrying
 
 from pkgpanda.build import load_json
 from test_util.marathon import get_test_app, get_test_app_in_docker
 
 
+TIMEOUT = 500
+
```
Comment (mellenburg) [inline packages/dcos-integration-test/extra/test_networking.py]: This test adds some non-trivial log output on top of what was already a pretty decent amount of logging. Could we please add a module level logger?
```
log = logging.getLogger(__name__)
```
And then migrate any statements from logging to use that log. This way we can crank up the verbosity of just this test for easy debugging

File: packages/dcos-integration-test/extra/test_networking.py [hunk 18]
```diff
@@ -1,37 +1,161 @@
...
 import requests
-
 import retrying
 
 from pkgpanda.build import load_json
 from test_util.marathon import get_test_app, get_test_app_in_docker
 
 
+log = logging.getLogger(__name__)
+log.setLevel(logging.INFO)
```
Comment (mellenburg) [inline packages/dcos-integration-test/extra/test_networking.py]: Don't leave this log.setLevel(logging.INFO) in this commit. We want this logger to inherit the root logging level (which is already at INFO)

File: packages/dcos-integration-test/extra/test_networking.py [hunk 19]
```diff
@@ -1,37 +1,165 @@
...
+        sp = stack.enter_context(dcos_api_session.marathon.deploy_and_cleanup(proxy_app, timeout=timeout))
+        cmd = '/opt/mesosphere/bin/curl -s -f -m 5 http://{}/test_uuid'.format(r.vipaddr)
+        returned_uuid = ensure_routable(cmd, sp)
+        log.debug('returned_uuid is: {}'.format(returned_uuid))
+    assert returned_uuid is not None
+    assert returned_uuid['test_uuid'] == app_uuid
+    r.log('PASSED')
+
+
+@pytest.mark.skipif(not lb_enabled(), reason='Load Balancer disabled')
```
Comment (mellenburg) [inline packages/dcos-integration-test/extra/test_networking.py]: Skipping a fixture doesnt make too much sense, you want this decorator around test_vip I think

File: packages/navstar/extra/dcos-navstar.service [hunk 20]
```diff
@@ -21,4 +21,7 @@ ExecStartPre=/opt/mesosphere/bin/bootstrap dcos-navstar
 ExecStartPre=/opt/mesosphere/bin/bootstrap dcos-minuteman
 ExecStartPre=/usr/bin/mkdir -p /var/lib/dcos/navstar/mnesia
 ExecStartPre=/usr/bin/mkdir -p /var/lib/dcos/navstar/lashup
+ExecStartPre=/usr/bin/env modprobe dummy
+ExecStartPre=-/usr/bin/env ip link add minuteman type dummy
```
Comment (lingmann) [inline packages/navstar/extra/dcos-navstar.service]: The number of ExecStartPre's here and now the addition of a `-` (fails are ignored) on this line suggest that we need to turn all of this into a proper startup script with error handling.

File: packages/navstar/extra/dcos-navstar.service [hunk 21]
```diff
@@ -21,4 +21,7 @@ ExecStartPre=/opt/mesosphere/bin/bootstrap dcos-navstar
 ExecStartPre=/opt/mesosphere/bin/bootstrap dcos-minuteman
 ExecStartPre=/usr/bin/mkdir -p /var/lib/dcos/navstar/mnesia
 ExecStartPre=/usr/bin/mkdir -p /var/lib/dcos/navstar/lashup
+ExecStartPre=/usr/bin/env modprobe dummy
```
Comment (asridharan) [inline packages/navstar/extra/dcos-navstar.service]: Why do we need this dummy interface? I thought setting up the local routing table was enough?

File: test_util/marathon.py [hunk 22]
```diff
@@ -11,6 +11,7 @@
 
 TEST_APP_NAME_FMT = 'integration-test-{}'
 REQUIRED_HEADERS = {'Accept': 'application/json, text/plain, */*'}
+log = logging.getLogger(__name__)
```
Comment (asridharan) [inline test_util/marathon.py]: Why are we changing the logger for the python tests? Just wondering why its part of this PR?

File: packages/navstar/extra/dcos-navstar.service (hunk 20 above)
Comment (asridharan) [inline packages/navstar/extra/dcos-navstar.service]: +1 on @lingmann comment on converting this to script. We should just have one `ExecStartPre` point it to a shell and handle all failure conditions in the shell. Would help us debug errors a lot better.

Comment (aeyakovenko) [inline packages/navstar/extra/dcos-navstar.service]: @asridharan @lingmann filed a jira for this, https://mesosphere.atlassian.net/browse/DCOS-13698
i think it can be handled in a separate pull request.  we don't have a "way to do this" yet afaik.  so it will take a little bit of thought to implement

File: test_util/marathon.py (hunk 22 above)
Comment (mellenburg) [inline test_util/marathon.py]: This is the idiom we are moving towards in all test infrastructure and allows this VIP test in particular to dynamically control logging in an idiomatic way

File: packages/navstar/extra/dcos-navstar.service (hunk 21 above)
Comment (aeyakovenko) [inline packages/navstar/extra/dcos-navstar.service]: we need it to mark the routes as `minuteman`, otherwise we can't reconcile

File: packages/navstar/extra/dcos-navstar.service (hunk 20 above)
Comment (alberts) [inline packages/navstar/extra/dcos-navstar.service]: +1 on taking care of this in a follow-up PR

**Comment by lingmann** [comment]
@aeyakovenko looks good. Ready for the train as soon as tests go green.

**Comment by lingmann** [comment]
@asridharan have your issues been addressed / can you update your PR?

File: packages/dcos-integration-test/extra/test_networking.py [hunk 23]
```diff
@@ -1,37 +1,170 @@
...
+            failed_tests.remove(test)
+            passed_tests.append(test)
+
+        threads.append(threading.Thread(target=run, args=([r])))
+    groups = [threads[i:i + maxthreads] for i in range(0, len(threads), maxthreads)]
+    for gs in groups:
+        for t in gs:
+            t.start()
+        for t in gs:
+            t.join()
```
Comment (jgehrcke) [inline packages/dcos-integration-test/extra/test_networking.py]: For next time :-) https://docs.python.org/dev/library/concurrent.futures.html#threadpoolexecutor

**Comment by asridharan** [comment]
LGTM !! 🚢 
The only concern I have is with the number of `ExecPre` commands we are setting up. But I guess we can handle them in a separate PR.

File: packages/dcos-integration-test/extra/test_networking.py [hunk 24]
```diff
@@ -1,37 +1,170 @@
...
+    def __str__(self):
+        return ("VipTest(vip={},vipaddr={},samehost={},"
+                "vipnet={},proxynet={})").format(self.vip, self.vipaddr, self.samehost, self.vipnet, self.proxynet)
+
+    def log(self, s, lvl=logging.DEBUG):
+        m = 'VIP_TEST {} {}'.format(s, self)
+        log.log(lvl, m)
+
+
+def vip_test(dcos_api_session, r):
```
Comment (asridharan) [inline packages/dcos-integration-test/extra/test_networking.py]: Should have pointed this out earlier, this test is very `DockerContainerizer` specific. Would be a pain to change it if you want to replicate it for UCR. Can we modify it so that the same vip_test can be invoked with a paramter to test `DockerContainerizer` vs UCR?

Comment (asridharan) [inline packages/dcos-integration-test/extra/test_networking.py]: @aeyakovenko sorry for the preemptive 🚢 wanted to check if the above is going to be a problem?

Comment (aeyakovenko) [inline packages/dcos-integration-test/extra/test_networking.py]: already have a jira and PR for UCR :) https://github.com/dcos/dcos/pull/1224

Comment (asridharan) [inline packages/dcos-integration-test/extra/test_networking.py]: Looks like the refactor will be done in another PR. so 🚢

Comment (orsenthil) [inline packages/dcos-integration-test/extra/test_networking.py]: [minor comments]:
* Please rename `r` to something meaningful 🔴 

* Also, please use a rough guideline of about 5 lines for the function. If you go by this, this function can see some love.

**Comment by asridharan** [comment]
@lingmann is it possible to indicate here which train this pull request is going to catch? I see Train 128 being mentioned here but that doesn't seem to include these commits?

**Comment by orsenthil** [comment]
@mesosphere-mergebot bump-ee

**Comment by alberts** [comment]
@lingmann bump-ee looks green now. let's :train2: ?

**Comment by lingmann** [comment]
Yay! Nice work @aeyakovenko.
